packages/mui-base/src/useSlider/useSlider.ts

```diff
@@ -140,6 +140,15 @@ function focusThumb({
   }
 }
 
+function areValuesEqual(
+  newValue: number | Array<number>,
+  oldValue: number | Array<number>,
+): boolean {
+  return typeof newValue === 'number' || typeof oldValue === 'number'
+    ? newValue === oldValue
+    : areArraysEqual(newValue, oldValue);
```

Comment:
Actually this should be for better checks:
```suggestion
  if (typeof newValue === 'number' && typeof oldValue === 'number') {
    return newValue === oldValue;
  }
  if (typeof newValue === 'object' && typeof oldValue === 'object') {
    return areArraysEqual(newValue, oldValue);
  }
  return false;
```